### PR TITLE
feat: implement holdings value calculation with price fallback

### PR DESCRIPTION
**Pull request by moltboie**
## Summary
Implements holdings value calculation with price fallback as specified in issue #26.

## Changes
- **`buildSecurityPriceIndex()`** - Index security prices by security_id and yearMonth
- **`getPriceForHolding()`** - 3-tier price fallback strategy:
  1. institution_price from holding (brokerage-reported)
  2. close_price from security snapshot (market data)
  3. Infer from institution_value / quantity
- **`inferCostBasis()`** - Calculate cost basis from buy transactions using average cost method
- **`getHoldingsValueData()`** - Main calculation function that builds HoldingsValueData from snapshots and transactions

## Files Changed
- `src/client/lib/hooks/calculation/holdings.ts` - New calculation module
- `src/client/lib/hooks/calculation/holdings.test.ts` - 18 unit tests
- `src/client/lib/hooks/calculation/index.ts` - Export new module
- `bunfig.toml` - Test preload configuration
- `src/client/lib/hooks/calculation/test-preload.ts` - Browser env setup for tests

## Testing
- TypeScript compiles cleanly (`npx tsc --noEmit`)
- All 53 holdings-related tests pass (`bun test holdings`)
- Tests cover:
  - Price fallback priority logic
  - Cost basis inference with average cost method
  - SELL transaction handling
  - Date and account/security filtering
  - Edge cases (null prices, empty data, etc.)

## Related
- Contributes to #26
- Part of Holdings Value Calculation design: https://doc.hoie.kim/p/ZDhNQZtVic

### CODE REVIEW

**Review by moltboie — COMMENTED** [review]
## Self-Review ✅

Reviewed holdings value calculation implementation (722 lines).

**Implementation (holdings.ts):**
- `buildSecurityPriceIndex`: Efficiently indexes security prices by security_id and yearMonth
- `getPriceForHolding`: Implements 3-tier fallback (institution → market → inferred) per design doc
- `inferCostBasis`: Average cost method with proper BUY/SELL handling
- `getHoldingsValueData`: Main orchestration function connecting all pieces

**Test Coverage (holdings.test.ts):**
- `buildSecurityPriceIndex`: 3 tests (basic indexing, null handling, empty dict)
- `getPriceForHolding`: 4 tests covering all 3 priorities and null case
- `inferCostBasis`: Tests for buy transactions, fees, sell transactions, mixed scenarios
- `getHoldingsValueData`: Integration tests for full flow

**Design Alignment:**
- Follows Holdings Value Design doc price fallback strategy
- Cost basis inference uses average cost method as specified
- Clean separation of concerns with exported pure functions

**Notes:**
- React hook intentionally kept separate (comment in code explains why)
- Type exports enable downstream consumers

**No issues found.** Implementation matches design doc.

File: package.json [hunk 1]
```diff
@@ -20,8 +20,10 @@
     "dev": "bun run dev-client & bun run dev-server",
     "dev-client": "vite",
     "dev-server": "bun --watch ./src/server/start.ts",
-    "test": "bun test",
-    "test:coverage": "bun test --coverage --coverage-reporter=lcov",
+    "test": "bun run test:client && bun run test:server",
+    "test:client": "bun test --preload ./src/client/test-setup.ts src/client src/common",
+    "test:server": "bun test src/server",
```
[the remark this answers is not in the document]
Comment (moltboie) [reply package.json]: Fixed in 346fa3c. `test:server` now includes `src/common`:

```
"test:server": "bun test src/server src/common"
```

Both `test:client` and `test:server` cover `src/common` so common utilities are tested in both environments.

**Review by hoiekim — APPROVED** [review]
